src/components/algorithm/AddAlgorithm.js

```diff
@@ -0,0 +1,234 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import { withTranslation } from 'react-i18next';
+import { withRouter } from 'react-router';
+import { connect } from 'react-redux';
+
+import Container from '@material-ui/core/Container';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import SaveIcon from '@material-ui/icons/Save';
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+import Radio from '@material-ui/core/Radio';
+import RadioGroup from '@material-ui/core/RadioGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import FormControl from '@material-ui/core/FormControl';
+import Grid from '@material-ui/core/Grid';
+
+import Main from '../common/Main';
+import BrowseFileButton from '../common/BrowseFileButton';
+import Editor from '../common/Editor';
+import { addAlgorithm } from '../../actions';
+import PYTHON_TEMPLATE_CODE from './pythonTemplateCode';
+import { AUTHOR_USER } from '../../config/constants';
+
+const styles = (theme) => ({
+  buttons: {
+    marginTop: theme.spacing(2),
+  },
+});
+
+const OPTION_FILE = 'OPTION_FILE';
+const OPTION_EDITOR = 'OPTION_EDITOR';
+
+class AddAlgorithm extends Component {
+  state = {
+    name: '',
+    description: '',
+    fileLocation: '',
+    code: PYTHON_TEMPLATE_CODE,
+    option: OPTION_FILE,
+  };
+
+  static propTypes = {
+    history: PropTypes.shape({
+      goBack: PropTypes.func.isRequired,
+    }).isRequired,
+    dispatchAddAlgorithm: PropTypes.func.isRequired,
+    t: PropTypes.func.isRequired,
+    classes: PropTypes.shape({
+      buttons: PropTypes.string.isRequired,
+    }).isRequired,
+  };
+
+  handleNameOnChange = (event) => {
+    this.setState({ name: event.target.value });
+  };
+
+  handleDescriptionOnChange = (event) => {
+    this.setState({ description: event.target.value });
+  };
+
+  handleLocationInput = (event) => {
+    this.setState({ fileLocation: event.target.value });
+  };
+
+  handleBrowseFileCallback = (filePaths) => {
+    // currently we select only one file
+    const filePath = filePaths[0];
+    if (filePath) {
+      this.setState({ fileLocation: filePath });
+    }
+  };
+
+  handleCodeOnChange = (code) => {
+    this.setState({ code });
+  };
+
+  handleOptionOnChange = (event) => {
+    this.setState({ option: event.target.value });
+  };
+
+  handleBack = () => {
+    const {
+      history: { goBack },
+    } = this.props;
+    goBack();
+  };
+
+  handleSave = () => {
+    const { dispatchAddAlgorithm } = this.props;
+    const { name, description, fileLocation, code, option } = this.state;
+    const author = AUTHOR_USER;
+    const payload =
+      option === OPTION_FILE
+        ? { name, description, author, fileLocation }
+        : { name, description, author, code };
+    if (name) {
```

Comment:
Maybe instead of only dispatching if there is a name, have a form control and highlight the errors. https://material-ui.com/components/checkboxes/#CheckboxesGroup.js 
Or have something simple as disabling the save button if the mandatory fields are not filled in. (Or have both)